metrics/metrics-core/src/main/java/io/servicecomb/metrics/core/publish/JsonMetricsPublisher.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.metrics.core.publish;
+
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.servicecomb.foundation.common.exceptions.ServiceCombException;
+import io.servicecomb.foundation.common.utils.JsonUtils;
+
+@Component
+public class JsonMetricsPublisher implements MetricsPublisher {
+
+  private final DataSource dataSource;
+
+  public JsonMetricsPublisher(DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
+
+  @Override
+  public String metrics(int pollerIndex) {
```

Comment:
It‘s not a good way to just return the string here.

Reply:
I had post a mail discuss it : https://lists.apache.org/thread.html/8c34a561c0c4e34b757e32e58c5a99e205c157326292d479526da4c3@%3Cdev.servicecomb.apache.org%3E